src/server/prisma/index.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * Instantiates a single instance PrismaClient and save it on the global object.
+ * @link https://www.prisma.io/docs/support/help-articles/nextjs-prisma-client-dev-practices
+ */
+import { PrismaClient } from '@prisma/client'
+import pino from 'pino'
+import { env } from '~/env.mjs'
+import { makePgliteClient, applyMigrations } from './pglite'
+
+const prismaGlobal = global as typeof global & {
+  prisma?: PrismaClient
+}
+
+const choosePrismaClient = () => {
+  if (
+    !env.DATABASE_URL &&
+    (env.NODE_ENV === 'development' || env.NODE_ENV === 'test')
+  ) {
+    pino().warn({}, 'DATABASE_URL not set, using pglite')
```

Comment:
Hmm allow pass in of pino instance instead of using global?

Reply:
There's no opportunity to do this, since callers expect a straight up export of `prisma: PrismaClient`. Given that the log happens just once, I feel it is a fair-enough compromise to invoke the pino global just once